python_modules/dagster/dagster/_core/definitions/definitions_class.py

```diff
@@ -94,10 +94,37 @@ def created_repo():
                 *(jobs or []),
             ]
 
-        self._created_repo = created_repo
+        self._created_pending_or_normal_repo = created_repo
+        self._resolved_repo: Optional[RepositoryDefinition] = None
 
-    def get_inner_repository(self) -> Union[RepositoryDefinition, PendingRepositoryDefinition]:
-        return self._created_repo
+    @public
+    def get_job_def(self, name: str) -> JobDefinition:
+        """Get a job definition by name. If you passed in a an UnresolvedAssetJobDefinition
+        (return value of define_asset_job) it will be resolved to a JobDefinition when returned
+        from this function."""
+
+        check.str_param(name, "name")
+        return self.get_repository_def().get_job(name)
+
+    def get_repository_def(self) -> RepositoryDefinition:
```

Comment:
if you used `@cached_method`, this could be:
```python
@cached_method
def get_repository_def(self) -> RepositoryDefinition:
    return (
        self._created_pending_or_normal_repo.compute_repository_definition()
        if isinstance(self._created_pending_or_normal_repo, PendingRepositoryDefinition)
        else self._created_pending_or_normal_repo
    )
```